File: AssistantPlus/Application/View/CoreResourceShipper/Configuration.cs
#pragma warning disable 0,
// ReSharper disable

using Twinning.AssistantPlus;
using Twinning.AssistantPlus.Utility;

namespace Twinning.AssistantPlus.View.CoreResourceShipper {

	public record PresetConfiguration {
		public String                     Name     { get; set; } = default!;
		public Dictionary<String, Object> Argument { get; set; } = default!;
	}

	public enum FilterType {
		Any,
		File,
		Directory,
	}

	public record FilterConfiguration {
		public FilterType Type { get; set; } = default!;
		public String     Name { get; set; } = default!;
	}

	public record OptionConfiguration {
		public String                     Name   { get; set; } = default!;
		public String                     Icon   { get; set; } = default!;
		public FilterConfiguration?       Filter { get; set; } = default!;
		public Boolean                    Batch  { get; set; } = default!;
		public String?                    Method { get; set; } = default!;
		public List<PresetConfiguration?> Preset { get; set; } = default!;
	}

	public record OptionGroupConfiguration {
		public String                    Name { get; set; } = default!;
		public String                    Icon { get; set; } = default!;
		public List<OptionConfiguration> Item { get; set; } = default!;
	}

	public record Configuration {
		public List<OptionGroupConfiguration> Option { get; set; } = default!;
	}

}

File: module/assistant/lib/view/core_resource_shipper/configuration.dart
import '/common.dart';

// ----------------

class PresetConfiguration {
  String              name;
  Map<String, Object> argument;
  PresetConfiguration({
    required this.name,
    required this.argument,
  });
}

enum FilterType {
  any,
  file,
  directory,
}

class FilterConfiguration {
  FilterType type;
  String     name;
  FilterConfiguration({
    required this.type,
    required this.name,
  });
}

class OptionConfiguration {
  String                     name;
  String                     icon;
  FilterConfiguration?       filter;
  Boolean                    batch;
  String?                    method;
  List<PresetConfiguration?> preset;
  OptionConfiguration({
    required this.name,
    required this.icon,
    required this.filter,
    required this.batch,
    required this.method,
    required this.preset,
  });
}

class OptionGroupConfiguration {
  String                    name;
  String                    icon;
  List<OptionConfiguration> item;
  OptionGroupConfiguration({
    required this.name,
    required this.icon,
    required this.item,
  });
}

class Configuration {
  List<OptionGroupConfiguration> option;
  Configuration({
    required this.option,
  });
}
